internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -1023,17 +1077,28 @@ func FlattenSiteConfigWindowsAppSlot(appSiteSlotConfig *web.SiteConfig, currentS
 		siteConfig.WorkerCount = int(*appSiteSlotConfig.NumberOfWorkers)
 	}
 
-	winAppStack := ApplicationStackWindows{
-		NetFrameworkVersion:  utils.NormalizeNilableString(appSiteSlotConfig.NetFrameworkVersion),
-		PhpVersion:           utils.NormalizeNilableString(appSiteSlotConfig.PhpVersion),
-		NodeVersion:          utils.NormalizeNilableString(appSiteSlotConfig.NodeVersion),
-		PythonVersion:        utils.NormalizeNilableString(appSiteSlotConfig.PythonVersion),
-		JavaVersion:          utils.NormalizeNilableString(appSiteSlotConfig.JavaVersion),
-		JavaContainer:        utils.NormalizeNilableString(appSiteSlotConfig.JavaContainer),
-		JavaContainerVersion: utils.NormalizeNilableString(appSiteSlotConfig.JavaContainerVersion),
+	winAppStack := ApplicationStackWindows{}
+
+	winAppStack.NetFrameworkVersion = pointer.From(appSiteSlotConfig.NetFrameworkVersion)
+	if currentStack == CurrentStackDotNetCore {
+		winAppStack.NetCoreVersion = pointer.From(appSiteSlotConfig.NetFrameworkVersion)
+	}
+	winAppStack.PhpVersion = pointer.From(appSiteSlotConfig.PhpVersion)
+	if winAppStack.PhpVersion == "" {
+		winAppStack.PhpVersion = PhpVersionOff
+	}
+	winAppStack.NodeVersion = pointer.From(appSiteSlotConfig.NodeVersion)     // TODO - Get from app_settings
+	winAppStack.PythonVersion = pointer.From(appSiteSlotConfig.PythonVersion) // This _should_ always be `""`
+	winAppStack.Python = currentStack == CurrentStackPython
+	winAppStack.JavaVersion = pointer.From(appSiteSlotConfig.JavaVersion)
+	switch pointer.From(appSiteSlotConfig.JavaContainer) {
+	case JavaContainerTomcat:
+		winAppStack.TomcatVersion = *appSiteSlotConfig.JavaContainerVersion
+	case JavaContainerEmbeddedServer:
+		winAppStack.JavaEmbeddedServer = true
 	}
 
-	siteConfig.WindowsFxVersion = utils.NormalizeNilableString(appSiteSlotConfig.WindowsFxVersion)
+	siteConfig.WindowsFxVersion = pointer.From(appSiteSlotConfig.WindowsFxVersion)
 	if siteConfig.WindowsFxVersion != "" {
 		// Decode the string to docker values
 		parts := strings.Split(strings.TrimPrefix(siteConfig.WindowsFxVersion, "DOCKER|"), ":")
```

Comment:
```suggestion
                parts := strings.Split(strings.TrimPrefix(siteConfig.WindowsFxVersion, "DOCKER|"), ":")
        if len(parts) == 2 {
            winAppStack.DockerContainerTag = parts[1]
            path := strings.Split(parts[0], "/")
            if len(path) > 1 {
                winAppStack.DockerContainerRegistry = path[0]
                winAppStack.DockerContainerName = strings.TrimPrefix(parts[0], fmt.Sprintf("%s/", path[0]))
            } else {
                winAppStack.DockerContainerName = path[0]
            }
        }
    }
```

line 1108 should be > 1, as it is the same code as in windows_web_app_scheme

Reply:
Good spot, thanks @CalvinRossSmith